api/app/registration/api.py

```diff
@@ -1,21 +1,175 @@
+    def put(self):
+        # update existing offer
+        args = self.req_parser.parse_args()
+        offer_id = args['id']
+        accepted = args['accepted']
+        rejected = args['rejected']
+        rejected_reason = args['rejected_reason']
+
+        offer = db.session.query(Offer).filter(Offer.id == offer_id).first()
+
+        if not offer:
+            return errors.OFFER_NOT_FOUND
+        else:
+            return
+
+        offerValues = Offer(
+            accepted = accepted,
+            rejected = rejected,
+            rejected_reason = rejected_reason
+            
+        )
+     
+        try:
+            offer = db.session.query(Offer).filter(Offer.id == offer_id).first()
+            offer.updated_at = datetime.now()
+            user_id = verify_token(request.headers.get('Authorization'))['id']
+
+            if offer and (not offer.user_id == user_id):
+                return errors.FORBIDDEN
+            else:
+                db.session.add(offerValues)
+                db.session.commit()
+
+        except:
+            LOGGER.error("Failed to add offer with id {}".format(args['id']))
+            return errors.ADD_OFFER_FAILED
+        return offer(offerValues), 201
+
+    @admin_required
+    def post(self):
+        args = self.req_parser.parse_args()
+        user_id = args['user_id']
+        email = g.current_user['email']
+        event_id = args['event_id']
+        offer_date = args['offer_date']
+        expiry_date = args['expiry_date']
+        payment_required = args['payment_required']
+        travel_award = args['travel_award']
+        accommodation_award = args['accommodation_award']
+
+        user = db.session.query(AppUser).filter(AppUser.email == email).first()
+
+        offerEntity = Offer(
+            user_id=user_id,
+            event_id=event_id,
+            offer_date=offer_date,
+            expiry_date=expiry_date,
+            payment_required=payment_required,
+            travel_award=travel_award,
+            accommodation_award=accommodation_award
+        )
+
+        db.session.add(offerEntity)
+
+        try:
+            db.session.add(offerEntity)
+            db.session.commit()
+            if user.email:
+                send_mail(recipient=user.email,
+                    subject='Offer from Deep Learning Indaba',
+                    body_text= OFFER_EMAIL_BODY.format( 
+                                user.user_title, user.firstname, user.lastname,
+                                user_id, event_id, offer_date, 
+                                expiry_date, payment_required, 
+                                travel_award, accommodation_award))
+
+                LOGGER.debug("Sent an offer email to {}".format(user.email))
+            else:
+                user.verified_email = True
+                try:
+                    db.session.commit()
+                except IntegrityError:
+                    LOGGER.error("Unable to verify email: {}".format(email))
```

Comment:
Remove this else block - we shouldn't be verifying email addresses in the Offer controller